api/interactives.go

```
@@ -113,14 +120,87 @@ func (api *API) GetInteractiveMetadataHandler(w http.ResponseWriter, req *http.R
 	WriteJSONBody(metadata, w, http.StatusOK)
 }
 
-func (api *API) UpdateInteractiveInfoHandler(w http.ResponseWriter, req *http.Request) {
-	// called from the importer to update vis info/state
+func (api *API) UpdateInteractiveHandler(w http.ResponseWriter, req *http.Request) {
+
+	// 1. Check body has some metadata and json decodes
+	ctx := req.Context()
+	if req.Body == nil {
+		http.Error(w, "Empty body recieved", http.StatusBadRequest)
+		log.Error(ctx, "Empty body recieved", ErrEmptyBody)
+		return
+	}
+	updateResp := models.InteractiveUpdated{}
+	b, _ := ioutil.ReadAll(req.Body)
```

should we ignore this error? and possibly move the defer up before this? 🤔